packages/compile-solidity/src/run.ts

```diff
@@ -1,14 +1,81 @@
-const debug = require("debug")("compile:run");
-const OS = require("os");
-const semver = require("semver");
-const Common = require("@truffle/compile-common");
-const { CompilerSupplier } = require("./compilerSupplier");
-const { zeroLinkReferences, formatLinkReferences } = require("./shims");
+import { zeroLinkReferences, formatLinkReferences } from "./shims";
+import debugModule from "debug";
+const debug = debugModule("compile:run");
+import OS = require("os");
+import semver from "semver";
+import { CompilerSupplier } from "./compilerSupplier";
+import type {
+  Compilation,
+  Source,
+  CompiledContract
+} from "@truffle/compile-common";
+import type Config from "@truffle/config";
+import * as Common from "@truffle/compile-common";
+
+type Targets = string[];
+
+type CompilerOutput = {
+  contracts: {
+    [path: string]: object;
+  };
+  sources: {
+    [path: string]: {
+      ast?: object;
+      legacyAST?: object;
+      id: number;
+    };
+  };
+  errors?: any[];
+};
+
+type ProcessAllSourcesArgs = {
+  sources: Common.Sources.Sources;
+  compilerOutput: CompilerOutput;
+  originalSourcePaths: any;
+  language: string;
+};
+
+type PrepareCompilerInputArgs = {
+  sources: Common.Sources.Sources;
+  targets: Targets;
+  language: string;
+  settings: any;
+  modelCheckerSettings: any;
```

Comment:
Can these be given a more specific type?

Reply:
I changed these to `object` for now as I'm not certain how accurate I can be about what will be in here to be honest.

Reply:
If this code doesn't depend on any properties of these (sorry, I didn't go look again to see if it does), then that's maybe fine? Alternatively you could consider typing anything that is opaque to this package as `unknown`, as that would force a consumer to declare a type for it before actually consuming it.

I'll leave the decision as to whether or not that is appropriate to you, however.

Reply:
Do we want to force the user to do that?

Reply:
I'd vote yes. Using `unknown` for code that hasn't been typed yet creates a sort of boundary line for unchecked types in our code. If we use `any`, the unchecked types propagate out far-and-wide, and it can become very difficult to reign it in later.

That said, if there's a difference of opinion here, I'm not staunch enough about this to block the entire changeover to TS just for this one concern.

Here's some reading on the subject, should you want to consider it further.
https://dmitripavlutin.com/typescript-unknown-vs-any/